src/internal/operations/backup_integration_test.go

```diff
@@ -191,67 +206,210 @@ func checkMetadataFilesExist(
+				)
+			}
+		})
 	}
+}
 
-	paths := []path.Path{}
-	pathsByRef := map[string][]string{}
+// ---------------------------------------------------------------------------
+// Incremental Item Generators
+// TODO: this is ripped from factory.go, which is ripped from other tests.
+// At this point, three variation of the sameish code in three locations
+// feels like something we can clean up.  But, it's not a strong need, so
+// this gets to stay for now.
+// ---------------------------------------------------------------------------
 
-	for _, fName := range files {
-		p, err := path.Builder{}.
-			Append(fName).
-			ToServiceCategoryMetadataPath(tenant, user, service, category, true)
-		if !assert.NoError(t, err, "bad metadata path") {
-			continue
-		}
+// the params here are what generateContainerOfItems passes into the func.
+// the callback provider can use them, or not, as wanted.
+type dataBuilderFunc func(id, now, subject, body string) []byte
 
-		dir, err := p.Dir()
-		if !assert.NoError(t, err, "parent path") {
-			continue
-		}
+//revive:disable:context-as-argument
+func generateContainerOfItems(
+	t *testing.T,
+	ctx context.Context,
+	gc *connector.GraphConnector,
+	service path.ServiceType,
+	cat path.CategoryType,
+	sel selectors.Selector,
+	tenantID, userID, destFldr string,
+	howMany int,
+	dbf dataBuilderFunc,
+) (*details.Details, error) {
+	//revive:enable:context-as-argument
+	t.Helper()
+
+	items := make([]incrementalItem, 0, howMany)
 
-		paths = append(paths, p)
-		pathsByRef[dir.ShortRef()] = append(pathsByRef[dir.ShortRef()], fName)
+	for i := 0; i < howMany; i++ {
+		var (
+			now       = common.Now()
+			nowLegacy = common.FormatLegacyTime(time.Now())
+			id        = uuid.NewString()
+			subject   = "incr_test " + now[:16] + " - " + id[:8]
+			body      = "incr_test " + cat.String() + " generation for " + userID + " at " + now + " - " + id
+		)
+
+		items = append(items, incrementalItem{
+			name: id,
+			data: dbf(id, nowLegacy, subject, body),
+		})
 	}
 
-	cols, err := kw.RestoreMultipleItems(ctx, bup.SnapshotID, paths, nil)
-	assert.NoError(t, err)
+	collections := []incrementalCollection{{
+		pathElements: []string{destFldr},
+		category:     cat,
+		items:        items,
+	}}
 
-	for _, col := range cols {
-		itemNames := []string{}
+	dest := control.DefaultRestoreDestination(common.SimpleTimeTesting)
+	dest.ContainerName = destFldr
 
-		for item := range col.Items() {
-			assert.Implements(t, (*data.StreamSize)(nil), item)
+	dataColls, err := buildCollections(
+		t,
+		service,
+		tenantID, userID,
+		dest,
+		collections)
+	if err != nil {
+		return nil, err
+	}
 
-			s := item.(data.StreamSize)
-			assert.Greaterf(
-				t,
-				s.Size(),
-				int64(0),
-				"empty metadata file: %s/%s",
-				col.FullPath(),
-				item.UUID(),
-			)
+	return gc.RestoreDataCollections(ctx, sel, dest, dataColls)
+}
 
-			itemNames = append(itemNames, item.UUID())
-		}
+type incrementalItem struct {
+	name string
+	data []byte
+}
```

Comment:
I could see the utility of this being in a global package, maybe even mocks or something. It works similarly to the `data.Stream`.  

Not necessary to fix. Just highlighting that within testing we continue to have to make these structs and if they just happened to be in the tool belt, well, they would be available for us within testing.

Reply:
Generally agreed.  Although, and I haven't reviewed all the variations recently, it's possible that each instance is configured to fulfill some specific needs.  That would provide sufficient utility to maintain the apparent duplication.